arcgis-ios-sdk-samples/Analysis/Statistical query (group and sort)/StatisticalQueryGroupAndSortViewController.swift

```diff
@@ -0,0 +1,424 @@
+            orderByFieldsViewController.popoverPresentationController?.sourceRect = sender.bounds
+
+            // Present view controller
+            self.present(orderByFieldsViewController, animated: true, completion: nil)
+        }
+    }
+    
+    // MARK: - Table View Data Source
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return sectionHeaderTitles.count
+    }
+    
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 44
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return sectionHeaderTitles[section]
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        //
+        // Create the view
+        let returnedView = UIView(frame: CGRect(x: 0, y: 0, width: tableView.frame.size.width, height: 44))
+        returnedView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        returnedView.backgroundColor = UIColor.primaryBlue()
+        returnedView.layer.borderColor = UIColor.white.cgColor
+        returnedView.layer.borderWidth = 1
+
+        // Add label
+        let label = UILabel(frame: CGRect(x: 10, y: 7, width: view.frame.size.width, height: 25))
+        label.text = sectionHeaderTitles[section]
+        label.textColor = UIColor.white
+        returnedView.addSubview(label)
+
+        // Add button
+        let headerButton = UIButton(type: .contactAdd)
+        headerButton.frame = CGRect(x: tableView.frame.size.width - 32, y: 11, width: 22, height: 22)
+        headerButton.tintColor = UIColor.white
+        headerButton.tag = section
+        headerButton.addTarget(self, action:  #selector(headerButtonAction(_:)), for: .touchUpInside)
+        returnedView.addSubview(headerButton)
+        headerButton.translatesAutoresizingMaskIntoConstraints = false
+        returnedView.addConstraint(NSLayoutConstraint(item: headerButton, attribute: .centerY, relatedBy: .equal, toItem: returnedView, attribute: .centerY, multiplier: 1, constant: 0))
+        returnedView.addConstraint(NSLayoutConstraint(item: headerButton, attribute: .trailing, relatedBy: .equal, toItem: returnedView, attribute: .trailing, multiplier: 1, constant: -10))
+
+        return returnedView
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == 0 {
+            return statisticDefinitions.count
+        }
+        else if section == 1 {
+            return selectedGroupByFieldNames.count
+        }
+        else {
+            return selectedOrderByFields.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        //
+        // Build the cell
+        let cell = tableView.dequeueReusableCell(withIdentifier: "TableViewCell")! as UITableViewCell
```

Comment:
Can use the overload that does not return an optional:

``` Swift
let cell = tableView.dequeueReusableCell(withIdentifier: "TableViewCell", for: indexPath)
```

Reply:
fixed